Server/fn_startTraffic.sqf

```diff
@@ -0,0 +1,282 @@
+               default {_roadSegments = ENGIMA_TRAFFIC_roadSegments select _currentInstanceIndex};
+            };
+
+            private _destinationSegment = _roadSegments select (floor (random (count _roadSegments)));
+            private _destinationPos = getPos _destinationSegment;
+
+            private _direction = ((_destinationPos select 0) - (getPos _spawnSegment select 0)) atan2 ((_destinationPos select 1) - (getposATL _spawnSegment select 1));
+            private _roadSegmentDirection = getDir _spawnSegment;
+
+            if ((_roadSegmentDirection < 0) || (_roadSegmentDirection > 360)) then {
+               _roadSegmentDirection = _roadSegmentDirection mod 360;
+            };
+
+            if ((_direction < 0) || (_direction > 360)) {
+               _direction = _direction mod 360;
+            };
+
+            private _testDirection = _direction - _roadSegmentDirection;
+
+            if ((_testDirection < 0) || (_testDirection > 360)) {
+               _testDirection = _testDirection mod 360;
+            };
+
+            if (_testDirection > 90 && _testDirection < 270) then {
+               _direction = _roadSegmentDirection + 180;
+            }
+            else {
+               _direction = _roadSegmentDirection;
+            };
+
+            (getPos _spawnSegment) params ["_posX", "_posY"];
+
+            _posX = _posX + 2.5 * sin (_direction + 90);
+            _posY = _posY + 2.5 * cos (_direction + 90);
+            private _pos = [_posX, _posY, 0];
+
+            // Create vehicle
+            private _vehicleType = _possibleVehicles select (floor (random (count _possibleVehicles)));
+
+            // Run spawn script and attach handle to vehicle
+            private _vehicleArray = [_pos,_vehicleType,_side] call enigmaTraffic_fnc_createRebelVehicle;
+            _vehicleArray params ["_vehicle", "_vehicleGroup"];
+            INFO_1("enigmaTraffic: vehicle is %1", _vehicle);
+
+            {[_vehicle] call _x;} forEach enigmaTraffic_fnc_vehicleSpawnHandler;
+
+            private _vehiclesCrew = units _vehicleGroup;
+            // Array - 0: created vehicle (Object), 1: all crew (Array of Objects), 2: vehicle's group (Group)
+            private _result = [_vehicle, _vehiclesCrew, _vehicleGroup];
+
+            // Name vehicle
+            if (isNil "enigmaTraffic_MilitaryTraffic_CurrentEntityNo") then {
+               enigmaTraffic_MilitaryTraffic_CurrentEntityNo = 0
+            };
+
+            _currentEntityNo = enigmaTraffic_MilitaryTraffic_CurrentEntityNo;
+            enigmaTraffic_MilitaryTraffic_CurrentEntityNo = enigmaTraffic_MilitaryTraffic_CurrentEntityNo + 1;
+
+            _vehicleVarName = "enigmaTraffic_MilitaryTraffic_Entity_" + str _currentEntityNo;
+            _vehicle setVehicleVarName _vehicleVarName;
+            _vehicle call compile format ["%1=_this;", _vehicleVarName];

```

Comment:
`call compile format` -> `missionNamespace setVariable`..
How old is this code? :D

Reply:
Its from A2

Reply:
Wie würde das dann aussehen, hast du ein Bsp.?

Reply:
missionNamespace setVariable [_vehicleVarName, _vehicle]